### PR TITLE
Logging migration

### PR DESCRIPTION
**Pull request by billsegall**
Replace github.com/op/go-logging with go.uber.org/zap

go-logging is unmaintained since 2016 and has significant re-entrancy problems
causing crashes. See for example:
op/go-logging#75
op/go-logging#102
op/go-logging#105

To fix this we introduce a wrapper for a subset of the go-logging library
mapped on to Uber's zap logging library and switch to using that.

### CODE REVIEW

File: archive/options.go [hunk 1]
```diff
@@ -16,7 +16,7 @@ package archive
 
 import (
 	"github.com/lirm/aeron-go/aeron/idlestrategy"
-	logging "github.com/op/go-logging"
+	"go.uber.org/zap/zapcore"
```
Comment (lirm) [inline archive/options.go:19]: Why not use the same abstraction here? Seems to be the only exception.

Comment (billsegall) [reply archive/options.go:19]: I could remove the archive options that hold the loglevel as they're currently unused. Let me know if you would prefer that for now.

It's my intention to provide methods from the archive to allow runtime control over the logging not only of the level (as foreshadowed) but also in the future to provide access to some further options that make use of the available zap runtime features.

This is for now a minimalist set of code changes across aeron-go that switch to zap. I imagined over time that you may want to adopt more of it generally and more directly but this should get over the immediate concurrency issues without large code changes.

**Comment by billsegall** [comment]
Was there something you would now have me change here?

Bill.

File: archive/options.go (hunk 1 above)
Comment (lirm) [reply archive/options.go:19]: I guess it's ok as a step in migration process and it is isolated.

**Pull request merged**